src/gui/sdlmain.cpp

```diff
@@ -2923,28 +2923,48 @@ static void show_warning(char const * const message) {
 #endif // WIN32
 }
 
-static void launcheditor() {
-	std::string path,file;
+static int LaunchEditor()
+{
+	std::string path, file;
 	Cross::CreatePlatformConfigDir(path);
 	Cross::GetPlatformConfigName(file);
 	path += file;
-	FILE* f = fopen(path.c_str(),"r");
+
+	FILE *f = fopen(path.c_str(), "r");
 	if (!f && !control->PrintConfig(path)) {
-		printf("tried creating %s. but failed.\n", path.c_str());
-		exit(1);
+		fprintf(stderr, "Tried creating '%s', but failed.\n", path.c_str());
+		return 1;
 	}
-	if(f) fclose(f);
-/*	if(edit.empty()) {
-		printf("no editor specified.\n");
-		exit(1);
-	}*/
-	std::string edit;
-	while(control->cmdline->FindString("-editconf",edit,true)) //Loop until one succeeds
-		execlp(edit.c_str(),edit.c_str(),path.c_str(),(char*) 0);
-	//if you get here the launching failed!
-	printf("can't find editor(s) specified at the command line.\n");
-	exit(1);
+	if (f)
+		fclose(f);
+
+	auto replace_with_process = [path](const std::string &prog) {
+		execlp(prog.c_str(), prog.c_str(), path.c_str(), (char *)nullptr);
+	};
+
+	std::string editor;
+	constexpr bool remove_arg = true;
+	// Loop until one succeeds
+	while (control->cmdline->FindString("--editconf", editor, remove_arg))
+		replace_with_process(editor);
+
+	while (control->cmdline->FindString("-editconf", editor, remove_arg))
+		replace_with_process(editor);
+
+	const char *env_editor = getenv("EDITOR");
+	if (env_editor)
+		replace_with_process(env_editor);
+
+	replace_with_process("nano");
+	replace_with_process("vim");
+	replace_with_process("vi");
+	replace_with_process("notepad.exe");
+
+	fprintf(stderr, "Can't find any text editors.\n");
+	fprintf(stderr, "You can override EDITOR env variable to use your preffered one.\n");
+	return 1;
 }
```

Comment:
Is it possible to return from this function before this? (I might not be seeing it)

Reply:
If I understand correctly how `execlp` works, then this function will never return on success, as process will be replaced with an editor process (and successfully closed editor leaves status 0 in shell). Tested this on Linux and macOS, but maybe I am missing something.

I decided to replace all `exit` calls for consistency, but maybe I this is too much and I should go back to `exit(1)` here?

Reply:
Suggest we stick with it; you've done more testing than me - and it's working as expected. I just wasn't seeing an explicitly 'return 0'.  

Saw this nuance "On Windows 10, `execlp` works fine, until I compile with -mwindows in MinGW. Then it just makes my program terminate with a zero exit code."

https://stackoverflow.com/questions/44105982/win32-c-api-alternative-to-broken-execl-family-of-functions, which we probably shouldn't worry about given the original MingGW environment (non-MSYS2) already has plenty of other issues.

Reply:
Just tested MSVC build, with a little bit surprising effects; the dosbox process exits immediately with successful status in all cases:

- it works using PowerShell, exactly as expected (I presume cmd.exe is the same and all other native Windows solutions)
- it works (!) using WSL (session started inside Microsoft's new Terminal app) - works correctly for Windows editors (`notepad`, `notepad++`), does not work for `vim` (which is visible only inside WSL Ubuntu 18.04) - starts `notepad++` instead.
- it works using MINGW64 Git bash shell, but only when asked for `notepad`. By default or when asked for `vim`, it fails spectacularly leaving unusable MINGW shell behind...

Overall it works better than I anticipated and works correctly with all native Windows methods, so I presume it's good enough.